mediarouter-compose/src/main/java/ch/srgssr/androidx/mediarouter/compose/MediaRouteControllerDialogViewModel.kt

```diff
@@ -0,0 +1,275 @@
+    }.stateIn(viewModelScope, WhileSubscribed(), null)
+
+    val title = combine(
+        mediaDescription,
+        playbackState,
+        _selectedRoute,
+    ) { mediaDescription, playbackState, selectedRoute ->
+        if (selectedRoute.presentationDisplayId != RouteInfo.PRESENTATION_DISPLAY_ID_NONE) {
+            application.getString(R.string.mr_controller_casting_screen)
+        } else if (playbackState == null || playbackState.state == PlaybackStateCompat.STATE_NONE) {
+            application.getString(R.string.mr_controller_no_media_selected)
+        } else if (mediaDescription?.title.isNullOrEmpty() && mediaDescription?.subtitle.isNullOrEmpty()) {
+            application.getString(R.string.mr_controller_no_info_available)
+        } else {
+            mediaDescription.title?.toString()
+        }
+    }.stateIn(viewModelScope, WhileSubscribed(), null)
+    val subtitle = mediaDescription.map { it?.subtitle?.toString() }
+        .stateIn(viewModelScope, WhileSubscribed(), null)
+    val iconInfo = playbackState.map { playbackState ->
+        if (playbackState == null) {
+            return@map null
+        }
+
+        val isPlaying = playbackState.state == PlaybackStateCompat.STATE_BUFFERING
+                || playbackState.state == PlaybackStateCompat.STATE_PLAYING
+
+        val icon: ImageVector
+        val contentDescription: String
+        if (isPlaying && playbackState.isPauseActionSupported) {
+            icon = Icons.Pause
+            contentDescription = application.getString(R.string.mr_controller_pause)
+        } else if (isPlaying && playbackState.isStopActionSupported) {
+            icon = Icons.Stop
+            contentDescription = application.getString(R.string.mr_controller_stop)
+        } else if (!isPlaying && playbackState.isPlayActionSupported) {
+            icon = Icons.PlayArrow
+            contentDescription = application.getString(R.string.mr_controller_play)
+        } else {
+            return@map null
+        }
+
+        icon to contentDescription
+    }.stateIn(viewModelScope, WhileSubscribed(), null)
+
+    init {
+        router.addCallback(
+            MediaRouteSelector.EMPTY,
+            mediaRouterCallback,
+            MediaRouter.CALLBACK_FLAG_UNFILTERED_EVENTS,
+        )
+
+        router.mediaSessionToken?.let { mediaSessionToken ->
+            val mediaController = MediaControllerCompat(application, mediaSessionToken)
+            mediaController.registerCallback(mediaControllerCallback)
+
+            this.mediaController = mediaController
+            mediaDescription.update { mediaController.metadata?.description }
+            playbackState.update { mediaController.playbackState }
+        }
+    }
+
+    fun hideDialog() {
+        savedStateHandle[KEY_SHOW_DIALOG] = false
+    }
+
+    fun toggleDeviceGroup() {
+        _isDeviceGroupExpanded.update { !it }
+    }
+
+    fun onStopCasting() {
```

Comment:
We could rename this method like that

```suggestion
    fun stopCasting() {
```

Maybe the other "callback" method should have better name?